cpp/api.cpp

```diff
@@ -206,3 +206,17 @@ LDClientCPP::unregisterFeatureFlagListener(const std::string &name,
 {
     LDClientUnregisterFeatureFlagListener(this->client, name.c_str(), fn);
 }
+
+bool
+LDClientCPP::registerFeatureFlagListenerUserData(const std::string &name,
+    LDlistenerUserDatafn fn, void *const userdata)
+{
+    return LDClientRegisterFeatureFlagListenerUserData(this->client, name.c_str(), fn, userdata);
+}
```

Comment:
I'm not 100% sure how this interacts with C++ lambdas. 

Referencing your original use-case:
```c++
void registerListener(MainThreadCallback callback) {
  auto* client = LDClientCPP::Get();
  client->registerFeatureFlagListenerWithClosure("foo",
    [](const char* const flagKey, const int status, void* closure) {
      // We're being called on a LaunchDarkly thread, let's redirect back to the main thread.
      MainThreadCallback mainThreadCallback = (MainThreadCallback)closure;
      DispatchToMainThread(mainThreadCallback);
  }, (void*)callback);
}
```
Is the function pointer that is actually passed into `registerFeatureFlagListenerWithClosure/UserData` the same for every invocation of `registerListener`, or does the lambda expression make a unique one each time?